prboom2/src/r_state.h

```diff
@@ -119,4 +119,33 @@ extern int              rw_angle1;
 extern visplane_t       *floorplane;
 extern visplane_t       *ceilingplane;
 
+// GL render-only structures
+extern vertex_t* gl_vertexes;
+extern int gl_numvertexes;
+
+extern seg_t* gl_segs;
+extern int gl_numsegs;
+
+extern seg_t* gl_rsegs;
+
+extern subsector_t* gl_subsectors;
+extern int gl_numsubsectors;
+
+extern byte* gl_map_subsectors;
+
+extern node_t* gl_nodes;
+extern int gl_numnodes;
+
+// Components, connected group of subsectors in a sector
+extern component_t* gl_components;
+extern unsigned int gl_numcomponents;
+
+// Bleed lists for components
+extern bleed_t* bleeds;
+extern unsigned int numbleeds;
+
+// Deferred components (used by gl_bsp.c)
+extern component_t** deferred;
```

Comment:
`deferred` is too generic a name for a global like this. I'd prefer to bundle related globals into structs to avoid more pollution. Maybe something like this:
```c
typedef struct {
  gl_component_t* gl_components;
  unsigned int gl_numcomponents;

  gl_bleed_t* bleeds;
  unsigned int numbleeds;

  gl_component_t** deferred;
  unsigned int numdeferred;
} gl_component_data_t;

extern gl_component_data_t gl_component_data;
```
There's probably a better name for this struct, you will have a better conceptual idea of what it represents.

Reply:
It's renamed `gl_deferred` for now, but I'll come up a better way of grouping things once we're confident in the functionality/performance of the code.